lib/pure/random.nim

```diff
@@ -635,20 +653,60 @@ when not defined(nimscript) and not defined(standalone):
       let time = int64(times.epochTime() * 1000) and 0x7fff_ffff
       result = initRand(time)
     else:
-      let now = times.getTime()
-      result = initRand(convert(Seconds, Nanoseconds, now.toUnix) + now.nanosecond)
+      proc getRandomState(): Rand =
+        when defined(nimscript):
+          result = Rand(
+            a0: CompileTime.hash.Ui,
+            a1: CompileDate.hash.Ui)
+          if not result.isValid:
+            result = DefaultRandSeed
+        else:
+          var urand: array[sizeof(Rand), byte]
+
+          for i in 0 .. 7:
+            if sysrand.urandom(urand):
+              copyMem(result.addr, urand[0].addr, sizeof(Rand))
+              if result.isValid:
+                break
+
+          if not result.isValid:
+            # When 2 processes executed at same time on different machines,
+            # `result` can still have same value.
+            let
+              pid = getCurrentProcessId()
+              t = getMonoTime().ticks
```

Comment:
If `urandom` failed 8 times in a row, just `quit` instead, don't continue after a bug.

Reply:
instead of quit, how about raise a new `RandError`, which user can then handle and use their preferred fallback; shouldn't be a concern in practice I'm assuming?

Reply:
And who knows about this lovely new exception. And how are you supposed to handle it? It's simply "we're screwed completely". The worst thing you can do after a bug is to continue after it because then you're in an unknown, untested state. That's a *very different* thing from a "restart after bug" as the restart starts in the program state that has seen most testing by design, the initial state.

Reply:
It `quit` when `urandom` failed 8 times in a row now.